python/adelphi/bin/adelphi

```diff
@@ -229,11 +229,15 @@ def contribute(ctx, token):
 
 
 @export.command()
+@click.option('--rampup-cycles', type=int, default=1000, help='Number of cycles to use for the nosqlbench rampup phase')
+@click.option('--main-cycles', type=int, default=1000, help='Number of cycles to use for the nosqlbench main phase')
```

Comment:
Added explicit CLI options to specify rampup and main cycles

Reply:
NoSQLBench supports template parameters such that you can override these values when launching the workload, which I think is preferred over adding more command-line args to Adelphi itself.

Reply:
Understood, but I don't see why this would be preferred over what's being done here.  Keep in mind that the goal is to re-use the same pattern we used with Gemini i.e. use the adelphi tool to create the config for another app and then call the app with that newly-generated config.  Under that scenario there is no logical or functional difference between specifying the args to nosqlbench at runtime vs. specifying the args to a pre-processor at runtime.

The approach I've taken here also has the (admittedly smaller) benefit of keeping the nosqlbench config simple making it easier to determine exactly what was run, something we couldn't do with a config that was significantly driven by runtime params.

Reply:
Particularly, I think adding options increases complexity, I usually like to ponder if something is really useful before adding just one more param. It's also one more thing we have to test (I don't think that has been covered in this PR).
Since we already have a way to set the ramp-up cycles through nb itself, I don't see the reason why we need to duplicate it on our end.

> Under that scenario there is no logical or functional difference between specifying the args to nosqlbench at runtime vs. specifying the args to a pre-processor at runtime.

That's exactly why I don't think we need the duplication.

>  to determine exactly what was run

That will be kept in helm's values.yaml, which can be exported with the results.

Reply:
If we agree that there's no logical difference between the two approaches then it seems to be entirely a cost/benefit trade-off.  And I guess my fundamental argument on that point is that the generated YAML is a _complete_ and  _self-contained_ description of the nosqlbench runs that were executed.  I don't have to consult other files (which hopefully were preserved) or worry about whether environmental information can be reconstructed; the file literally has everything I need.  I can take it and run it immediately in my local nosqlbench instance.  I can preserve it as a build artifact in Jenkins (or something similar) and know I have a way to re-create my nosqlbench run forevermore.  I'd argue it's difficult to overstate the value of such a feature for debugging/auditing/reproducability arguments.

The exchange I have to make to get this feature is the addition of two (maybe three) CLI arguments.  While this does represent some increase in complexity I'd argue that it is much closer to a small increase than a medium one... and nowhere near a large increase.  I'd also observe that your suggestion that a user could reconstruct the info necessary to re-create a nosqlbench run via the nosqlbench runtime args + values.yaml _also_ represents an increase in complexity... one might even argue a larger increase than what's done here.

Finally, I'm not sure I'd refer to this as a duplication of functionality in nosqlbench.  Our YAML generator takes some number of args and uses the value of those args to generate some of the YAML... that's not redundant to the nosqlbench support for parameterizing values, nor would I want it to be.

In the end I just don't see a clear benefit to leaving in the parameters for nosqlbench... it just doesn't seem to buy us nearly enough.

Reply:
I was referring to usability complexity. This will be available in the adelphi lib and people may use it standalone. Figuring out and setting a bunch of parameters in Adelphi increases friction.
On our side, it's an extra param to test and inconsistent with other params we already have to set through nosqlbench (scenario name).

I'll open a separate ticket so we can add the cycles arg to the nb template and unblock this PR.